core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -655,7 +662,7 @@ protected void parse() throws ProtocolException {
         lockTime = readUint32();
         optimalEncodingMessageSize += 4;
 
-        if(getVersionShort() >= 3 && getType() != Type.TRANSACTION_NORMAL) {
+        if(getVersionShort() >= SPECIAL_VERSION && getType() != Type.TRANSACTION_NORMAL) {
```

Comment:
You didn't mention this one, but I think `SPECIAL_VERSION` should be used here as well instead of `MAX_STANDARD_VERSION`?